aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -54,6 +54,34 @@ const OptionalScalarRef max) {
   build_unary_op(maybe_get_output(), self.to(result_type));
 }
 
+TORCH_META_FUNC2(clamp, Tensor) (
+const Tensor& self,
+const OptionalTensorRef min,
+const OptionalTensorRef max) {
+  TORCH_CHECK(min || max, "torch.clamp: At least one of 'min' or 'max' must not be None");
+  //Manual type promotion, since scalars have to participate in it
+  ScalarType result_type = self.scalar_type();
+  TORCH_CHECK(!isComplexType(result_type), "clamp is not supported for complex types");
+  //Floating is the highest supported
+  #define CLAMP_CONFIG()                    \
+    TensorIteratorConfig()                  \
+      .set_check_mem_overlap(true)          \
+      .add_output(maybe_get_output())       \
+      .add_input(self)                      \
+      .promote_inputs_to_common_dtype(true) \
+      .cast_common_dtype_to_outputs(true)   \
+      .enforce_safe_casting_to_output(true)
```

Comment:
I feel like a lambda should have worked here? Returning the config

Reply:
It has `C10_DISABLE_COPY_AND_ASSIGN(TensorIteratorConfig);` so I didn't find the right incantation.